sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,66 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
+
+keys() ->
+    [ {"foo.setting", fun foo__setting/1}
+    , {"foo.endpoint", fun foo__endpoint/1}
+    ].
+
+foo__setting(mapping) -> "app_foo.setting";
+foo__setting(type) -> string();
+foo__setting(converter) -> converter(string);
+foo__setting(validator) -> type_checker(foo__setting(type));
```

Comment:
I think it always makes sense to validate the type. (In order to omit this check, one can use `any()` type). Therefore, call to `typerefl:type_check` function should be made by the "framework" code, and `validator` clause should be optional, and reserved for custom validators. 
This will reduce the amount of boilerplate in the schema.

Reply:
agree to @k32 .
if validator clause returns a `fun`, it applies on top of typerefl check result.